pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2743,6 +2747,28 @@ private void updateTopicMessageDispatchRate() {
         });
     }
 
+    private void updateDispatchPauseOnAckStatePersistentEnabled() {
+        this.pulsar().getExecutor().execute(() -> {
+            forEachTopic(topic -> {
+                if (topic instanceof PersistentTopic) {
+                    // Update policies.
+                    PersistentTopic persistentTopic = (PersistentTopic) topic;
+                    persistentTopic.updateDispatchPauseOnAckStatePersistentEnabled();
+                    // Trigger new read if subscriptions has been paused before.
+                    if (!pulsar().getConfiguration().isDispatcherPauseOnAckStatePersistentEnabled()) {
+                        persistentTopic.updateDispatchPauseOnAckStatePersistentEnabled();
+                        persistentTopic.getSubscriptions().forEach((sName, subscription) -> {
+                            if (subscription.getDispatcher() == null) {
+                                return;
+                            }
+                            subscription.getDispatcher().afterAckMessages(null, 0);
+                        });
```

Comment:
would it make sense to hide this logic in a method in PersistentTopic?

Reply:
Ah, great suggestion. Improved.